platform/features/timeline/src/controllers/swimlane/TimelineSwimlaneDropHandler.js

```diff
@@ -129,17 +155,12 @@ define(
                 });
             }
 
-            // Check if a compose action is allowed for the object in this
-            // swimlane (we handle the link differently to set the index,
-            // but check for the existence of the action to invole the
-            // relevant policies.)
-            function allowsCompose(swimlane, domainObject) {
-                var actionCapability =
-                    swimlane.domainObject.getCapability('action');
-                return actionCapability && actionCapability.getActions({
-                    key: 'compose',
-                    selectedObject: domainObject
-                }).length > 0;
+            function drop(domainObject, targetObject, indexOffset) {
+                var action = chooseAction(targetObject, domainObject);
+                return action.perform().then(function () {
```

Comment:
no-op seems an unusual way of handling this case, rather than returning `undefined` from `chooseAction` and wrapping in q.when ?